src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -85,6 +85,22 @@ internal sealed partial class DocumentProvider : ForegroundThreadAffinitizedObje
             Marshal.ThrowExceptionForHR(runningDocumentTableForEvents.AdviseRunningDocTableEvents(new RunningDocTableEventsSink(this), out _runningDocumentTableEventCookie));
         }
 
+        [Obsolete("Use the overload with 'isAdditionalFile' instead", error: true)]
+        public IVisualStudioHostDocument TryGetDocumentForFile(
+            IVisualStudioHostProject hostProject,
+            string filePath,
+            SourceCodeKind sourceCodeKind,
+            Func<ITextBuffer, bool> canUseTextBuffer,
+            Func<uint, IReadOnlyList<string>> getFolderNames,
+            EventHandler updatedOnDiskHandler = null,
+            EventHandler<bool> openedHandler = null,
+            EventHandler<bool> closingHandler = null)
+        {
+            return TryGetDocumentForFile(
+                hostProject, filePath, sourceCodeKind, canUseTextBuffer, getFolderNames, isAdditionalFile: false,
+                updatedOnDiskHandler: updatedOnDiskHandler, openedHandler: openedHandler, closingHandler: closingHandler);
+        }
```

Comment:
It's tracked in https://github.com/dotnet/roslyn/issues/20812, and I will remove it in 15.5 as soon as TS removes their call to it.  Why does the presence of `Obsolete`d methods make editing Roslyn any more scary - the risk is the same...